xdsl/irdl/constraints.py

```diff
@@ -55,7 +56,14 @@ def update(self, other: ConstraintContext):
 
 _AttributeCovT = TypeVar("_AttributeCovT", bound=Attribute, covariant=True)
 
-ResolveType: TypeAlias = Attribute | Sequence[Attribute]
+ResolveType = Attribute | Sequence[Attribute]
```

Comment:
This is inferred but also doesn't hurt, makes sure that the RHS is purely in type land
```suggestion
ResolveType: TypeAlias = Attribute | Sequence[Attribute]
```